nav2-usability/scenario-code/dynamic-linking/dynamic_linking_beamer.dart

```diff
@@ -0,0 +1,171 @@
+import 'dart:math';
+
+import 'package:flutter/material.dart';
+import 'package:beamer/beamer.dart';
+
+void main() {
+  runApp(WishlistApp());
+}
+
+class Wishlist {
+  final String id;
+
+  Wishlist(this.id);
+}
+
+class AppState extends ChangeNotifier {
+  final List<Wishlist> wishlists = <Wishlist>[];
+
+  Wishlist? _selectedWishlist;
+
+  Wishlist? get selectedWishlist => _selectedWishlist;
+
+  set selectedWishlist(Wishlist? w) {
+    _selectedWishlist = w;
+    notifyListeners();
+  }
+
+  void addWishlist(Wishlist wishlist) {
+    wishlists.add(wishlist);
+    notifyListeners();
+  }
+}
+
+class WishlistLocation extends BeamLocation {
+  WishlistLocation(AppState appState, BeamState state)
```

Comment:
This is either a typo and you meant to write `beamState` or you are more experienced in beamer than myself, because it really **should** be called `beamerState`! :smile:
It is the one generated by router delegate right after parsing; the pure URI description that each `BeamLocation` will take, detect path and other parameters in it and copy it configured for itself, making it then a "`beamState`", i.e. its `state`.

I'm not sure will renaming it to `beamerState` bring more confusion, but `beamState` sounds like a good change. I can commit if you agree.

Speaking of this; I've been thinking a lot about this distinction between `BeamLocation.state` (the one you see in `update`) and `Beamer.state` (the one you see in `locationBuilder` and, consequently, in `BeamLocation`'s constructor); whether they should be represented by a different class or not. The purpose of each is to be extendable and provide a custom declarative routing. The final decision about this is yet to be made :)

Reply:
Hey @johnpryan 
Do you agree that I rename it to `beamState`?

Reply:
Sounds good to me!